pkg/resource/statefulset.go

```diff
@@ -30,23 +31,27 @@ type StatefulSetBuilder struct {
 	Selector labels.Labels
 }
 
-func (b StatefulSetBuilder) Build() (runtime.Object, error) {
-	ss := &appsv1.StatefulSet{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: b.StatefulSetName(),
+func (b StatefulSetBuilder) Build(obj runtime.Object) error {
+	ss, ok := obj.(*appsv1.StatefulSet)
+	if !ok {
+		return errors.New("failed to access StatefulSet object")
```

Comment:
nit: access is a bit misleading here. Maybe just `failed to cast to StatefulSet`?